server/src-lib/Hasura/RQL/DDL/Schema/Table.hs

```diff
@@ -440,19 +450,25 @@ runSqlP2 (RunSQL t cascade) = do
   -- refresh the gCtxMap in schema cache
   refreshGCtxMapInSchema
 
-  return $ encode (res :: RunSQLRes)
+  return res
 
+isAltrOrDrop :: QErrM m => T.Text -> m Bool
+isAltrOrDrop = either throwErr return . matchRegex regex False
   where
-    rawSqlErrHandler :: Q.PGTxErr -> QErr
-    rawSqlErrHandler txe =
-      let e = err400 PostgresError "query execution failed"
-      in e {qeInternal = Just $ toJSON txe}
+    throwErr s = throw500 $ "compiling regex failed: " <> T.pack s
+    regex = "alter table|drop table|alter view|drop view|replace view"
```

Comment:
This regex will catch all SQL having `alter`, `drop` and `replace` words. We explicitly need `alter table`, `replace view` etc.

Reply:
Our metadata can get inconsistent by dropping schemas/constraints too. Instead of explicitly listing all the objects it will be simpler to just use `alter|drop|replace`.